### PR TITLE
[FEATURE BRANCH] Fixes type registration and post-start-hook failures

### PR DESCRIPTION
**Pull request by yue9944882**
-->
```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/server/config.go [hunk 1]
```diff
@@ -553,8 +553,8 @@ func (c completedConfig) New(name string, delegationTarget DelegationTarget) (*G
 		}
 	}
 
-	if feature.DefaultFeatureGate.Enabled(features.RequestManagement) {
-		requestManagementHookName := "generic-apiserver-request-management"
+	requestManagementHookName := "generic-apiserver-request-management"
+	if feature.DefaultFeatureGate.Enabled(features.RequestManagement) && !s.isPostStartHookRegistered(requestManagementHookName) {
```
Comment (MikeSpreitzer) [inline staging/src/k8s.io/apiserver/pkg/server/config.go]: If this change is needed then there may be bigger problems.

**Comment by MikeSpreitzer** [comment]
@MikeSpreitzer

File: staging/src/k8s.io/apiserver/pkg/server/config.go (hunk 1 above)
Comment (yue9944882) [reply staging/src/k8s.io/apiserver/pkg/server/config.go]: > 	if c.SharedInformerFactory != nil && !s.isPostStartHookRegistered(genericApiServerHookName) {

i see the other hooks are doing the same.

**Comment by yue9944882** [comment]
@MikeSpreitzer let's get this in?

**Pull request merged**